butane_core/src/codegen/mod.rs

```diff
@@ -280,45 +285,39 @@ fn fields(ast_struct: &ItemStruct) -> impl Iterator<Item = &Field> {
     ast_struct.fields.iter()
 }
 
-fn get_option_sql_type(ty: &syn::Type) -> Option<DeferredSqlType> {
-    get_type_argument(ty, &OPTION_TYNAMES).map(|path| {
-        let inner_ty: syn::Type = syn::TypePath {
-            qself: None,
-            path: path.clone(),
+/// Extract the path from a type.
+///
+/// Requires that the type is a `TypePath` (e.g. `i8` or `Foo<T>`).
+pub fn extract_path_from_type(ty: &syn::Type) -> &syn::Path {
+    if let syn::Type::Path(type_path) = ty {
+        if type_path.qself.is_some() {
+            panic!("TypePath with qself is not supported by butane");
         }
-        .into();
+        return &type_path.path;
+    } else {
+        panic!("butane codegen only supports TypePath types");
+    }
+}
```

Comment:
Happy to return a `Result` if there is any chance that a `qself` could be valid in the definition of a model, field-type, etc that butane comes into context with.